### ISSUE TITLE
[new] PHPUnit - Allow assertEquals for object comparison

### ISSUE THREAD

**Issue by bram123** [kalessil/phpinspectionsea#45]
When writing unit tests we often assert if objects retrieved from different places are the same.
assertSame will only succeed when the two variables reference the same object.

But assertEquals will succeed when the two instances have all the same property values.

When the two variables are both objects, and instances from the same class. It would be useful to allow the assertEquals method.

Note: Also need this functionality for variables marked as class array (Object[]).

**Comment by rentalhost** [comment]
I agree in part. 

Maybe the dev intention is really check if actual result is the same object as expected, then is important keeping using the `static::assertSame()`. But, we could consider a _small fix_ where you can tell to inspection that do you know that objects are differents but expects the same properties.

In this case, I guess that the previous element should be a `static::assertNotSame()` that pass exactly the same parameters. The result should be:

```php
static::assertNotSame($a, $b);  // <-- The fix/quick-fix.
static::assertEquals($a, $b);   // <-- Your initial code.
```

**Comment by kalessil** [comment]
Done, fresh binary pushed to master - would you please give it a try and share feedback on the fix?

**Comment by kalessil** [comment]
@rentalhost : FYI, the fix skips reporting for incomplete types, array types, and objects.
Anyway theassertSame/assertNotSame is deactivated by default,
